src/phpDocumentor/Descriptor/Class_.php

```diff
@@ -0,0 +1,223 @@
+<?php
+/**
+ * phpDocumentor
+ *
+ * PHP Version 5.3
+ *
+ * @copyright 2010-2014 Mike van Riel / Naenius (http://www.naenius.com)
+ * @license   http://www.opensource.org/licenses/mit-license.php MIT
+ * @link      http://phpdoc.org
+ */
+
+namespace phpDocumentor\Descriptor;
+use phpDocumentor\Reflection\DocBlock;
+use phpDocumentor\Reflection\Element;
+use phpDocumentor\Reflection\Fqsen;
+
+/**
+ * Descriptor representing a Class.
+ */
+final class Class_ implements Element
+{
+    /**
+     * @var Fqsen Full Qualified Structural Element Name
+     */
+    private $fqsen;
+
+    /**
+     * @var DocBlock | Null
+     */
+    private $docBlock = null;
+
+    /** @var boolean $abstract Whether this is an abstract class. */
+    private $abstract = false;
+
+    /** @var boolean $final Whether this class is marked as final and can't be subclassed. */
+    private $final = false;
+
+    /**
+     * @var Class_ $parent The class this class is extending.
+     */
+    private $parent = null;
+
+    /** @var Interface_[] $implements References to interfaces that are implemented by this class. */
+    private $implements = array();
+
+    /** @var Constant[] $constants References to constants defined in this class. */
+    private $constants = array();
+
+    /** @var Property[] $properties References to properties defined in this class. */
+    private $properties = array();
+
+    /** @var Method[] $methods References to methods defined in this class. */
+    private $methods = array();
+
+    /** @var Trait_[] $usedTraits References to traits consumed by this class */
+    private $usedTraits = array();
+
+    /**
+     * Initializes the all properties representing a collection with a new Collection object.
+     *
+     * @param Fqsen $fqsen
+     * @param DocBlock $docBlock
+     * @param bool $abstract
+     * @param bool $final
+     */
+    public function __construct(Fqsen $fqsen, DocBlock $docBlock = null, Class_ $parent = null, $abstract = false, $final = false)
+    {
+        $this->fqsen = $fqsen;
+        $this->parent = $parent;
+        $this->docBlock = $docBlock;
+        $this->abstract = $abstract;
+        $this->final = $final;
+    }
+
+    /**
+     * Returns true when this method is final. Otherwise returns false.
+     *
+     * @return bool
+     */
+    public function isFinal()
+    {
+        return $this->final;
+    }
+
+    /**
+     * Returns true when this method is abstract. Otherwise returns false.
+     *
+     * @return bool
+     */
+    public function isAbstract()
+    {
+        return $this->abstract;
+    }
+
+    /**
+     * Returns the Class_ this class is extending if available.
+     *
+     * @return NUll|Class_
```

Comment:
This method should return a FQSEN object, see the constructor comment for more information